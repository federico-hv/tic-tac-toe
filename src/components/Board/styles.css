
.board {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: space-between;
	width: 100%;
	height: 100%;
}

.squares-container {
	display: flex;
	flex-wrap: wrap;
	align-items: center;
	justify-content: center;
	width: 300px;
}

.player-turn-1 {
	color: white;
}

.player-turn-2 {
	color: black;
}

.result {
	color: #D0A1FF;
}
