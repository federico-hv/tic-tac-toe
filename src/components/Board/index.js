import React from 'react';
import Piece from '../Piece';
import './styles.css';

const Board = ({data, onMove} ) => (
  <div className="board">
    {data.map((row, i) => row.map((el, j) => <Piece key={j} onSelect={(s)=>onMove([i,j])} symbol={el} />))}
  </div>
);

export default Board;