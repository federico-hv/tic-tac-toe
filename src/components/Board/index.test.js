import React from 'react';
import { mount, shallow } from 'enzyme';
import Board from './index';
import toJson from 'enzyme-to-json';


describe('<Board />', () => {

	it('Should a div with class board', ()=> {
		const data = [['X', '', 'O'],['O','X',''],['X','O','X']]
		const board = shallow(<Board data={data} />);

		expect(board.find('.board').length).toEqual(1);
	})

	it('Matches snapshot', () => {
		const data = [['X', '', 'O'],['O','X',''],['X','O','X']]
		const board = mount(<Board data={data} />);
		
		expect(toJson(board)).toMatchSnapshot();
		board.unmount();
	});
})