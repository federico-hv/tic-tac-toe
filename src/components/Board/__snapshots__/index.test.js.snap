// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> Matches snapshot 1`] = `
<Board
  data={
    Array [
      Array [
        "X",
        "",
        "O",
      ],
      Array [
        "O",
        "X",
        "",
      ],
      Array [
        "X",
        "O",
        "X",
      ],
    ]
  }
>
  <div
    className="board"
  >
    <Piece
      key="0"
      onSelect={[Function]}
      symbol="X"
    >
      <div
        className="piece"
        onClick={null}
      >
        X
      </div>
    </Piece>
    <Piece
      key="1"
      onSelect={[Function]}
      symbol=""
    >
      <div
        className="piece"
        onClick={[Function]}
      />
    </Piece>
    <Piece
      key="2"
      onSelect={[Function]}
      symbol="O"
    >
      <div
        className="piece"
        onClick={null}
      >
        O
      </div>
    </Piece>
    <Piece
      key="0"
      onSelect={[Function]}
      symbol="O"
    >
      <div
        className="piece"
        onClick={null}
      >
        O
      </div>
    </Piece>
    <Piece
      key="1"
      onSelect={[Function]}
      symbol="X"
    >
      <div
        className="piece"
        onClick={null}
      >
        X
      </div>
    </Piece>
    <Piece
      key="2"
      onSelect={[Function]}
      symbol=""
    >
      <div
        className="piece"
        onClick={[Function]}
      />
    </Piece>
    <Piece
      key="0"
      onSelect={[Function]}
      symbol="X"
    >
      <div
        className="piece"
        onClick={null}
      >
        X
      </div>
    </Piece>
    <Piece
      key="1"
      onSelect={[Function]}
      symbol="O"
    >
      <div
        className="piece"
        onClick={null}
      >
        O
      </div>
    </Piece>
    <Piece
      key="2"
      onSelect={[Function]}
      symbol="X"
    >
      <div
        className="piece"
        onClick={null}
      >
        X
      </div>
    </Piece>
  </div>
</Board>
`;
